components/logins/src/login.rs

```diff
@@ -367,7 +367,11 @@ pub struct SecureLoginFields {
 
 impl SecureLoginFields {
     pub fn encrypt(&self, encdec: &EncryptorDecryptor) -> Result<String> {
-        encdec.encrypt_struct(&self)
+        encdec.encrypt_struct(&self, "encrypt SecureLoginFields")
```

Comment:
nit suggestion: Maybe another way of ensuring that the `action` is set properly is to introduce a trait `Encryptable`  

```rs
trait Encryptable: Serializable {
  fn action_name() -> &'static str;

  fn encrypt(&self, encdec: &EncryptorDecryptor) -> Result<String> {
      encdec.encrypt_struct(&self)
  }
}

```

And changing the signature of `encrypt_struct` to

```rs
fn encrypt_struct<E: Encryptable>(&self, fields: &E) -> Result<String> {
   let name = fields.action_name();
   ....
} 
```

Thoughts?